lib/flake-utils.nix

```
@@ -0,0 +1,66 @@
+{ systems }:
+rec {
+  /* mapDefaultSystems using default supported systems
+     Usage in a nix flake:
+       {
+         description = "Flake demo";
+
+         inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+         outputs = { self, nixpkgs }:
+           nixpkgs.lib.mapDefaultSystems self.inputs (inputs:
```

Maybe this variant would be interesting too?

```suggestion
         outputs = nixpkgs.lib.mapDefaultSystems (inputs:
```

---

I like it, but I wonder if this is a bad role model, if what this article has a point https://blog.ysndr.de/posts/internals/2021-01-01-flake-ification/
I looked up the reference there, didn't see Eelco mentioning issues with non-explicit outputs arguments.
But I may have missed it.

```
Another issue you might come across:

{...}:
let
...
in {
  inherit shell app some-other-app;
}
How do you know what to import, what does it do? There are simply no inherent semantics and also no conventions.

Additionally to this Eelco also lists some more issues with this. Firstly, due to missing strong conventions and semantics composing packages is difficult, or not nice of an experience for the user. It also means that discovering packages can not be automated easily.
```